src/LibraryViewExtension/Handlers/ResourceHandlerFactory.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using CefSharp;
+
+namespace Dynamo.LibraryUI.Handlers
+{
+    /// <summary>
+    /// Implements IResourceHandlerFactory to provide resource handler for a given request
+    /// </summary>
+    class ResourceHandlerFactory : DefaultResourceHandlerFactory
+    {
+        private Dictionary<string, IResourceProvider> resourceProviders = new Dictionary<string, IResourceProvider>();
+        private HashSet<string> supportedSchemes = new HashSet<string>();
+
+        /// <summary>
+        /// This method is called before a resource is loaded, if a valid resource handler
+        /// is returned then this resource handler would serve the response of web request.
+        /// </summary>
+        public override IResourceHandler GetResourceHandler(IWebBrowser browserControl, IBrowser browser, IFrame frame, IRequest request)
+        {
+            try
+            {
+                IResourceHandler handler;
+                if(!Handlers.TryGetValue(request.Url, out handler))
+                {
+                    handler = this.GetResourceHandler(request);
+                }
+
+                return handler;
+            }
+            finally
+            {
+                request.Dispose();
```

Comment:
That's a good point, but this is how the [base class is implemented](https://github.com/cefsharp/CefSharp/blob/master/CefSharp/DefaultResourceHandlerFactory.cs#L79) so I tried to maintain it.